lib/util/security_manager.dart

```diff
@@ -0,0 +1,79 @@
+import 'dart:io';
+import 'package:aewallet/application/settings/language.dart';
+import 'package:aewallet/application/settings/theme.dart';
+import 'package:aewallet/infrastructure/datasources/hive_preferences.dart';
+import 'package:aewallet/model/available_language.dart';
+import 'package:aewallet/ui/util/ui_util.dart';
+import 'package:aewallet/ui/widgets/components/dialog.dart';
+import 'package:aewallet/util/case_converter.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter_gen/gen_l10n/localizations.dart';
+import 'package:flutter_jailbreak_detection/flutter_jailbreak_detection.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+class SecurityManager {
+  factory SecurityManager() {
+    return _singleton;
+  }
+
+  SecurityManager._internal();
+  static final SecurityManager _singleton = SecurityManager._internal();
+
+  Future<bool> isDeviceJailbroken() async {
+    return FlutterJailbreakDetection.jailbroken;
+  }
+
+  Future<bool> isDeviceDeveloperMode() async {
+    return FlutterJailbreakDetection.developerMode;
+  }
+
+  Future<bool> isDeviceSecured() async {
+    return await SecurityManager().isDeviceJailbroken() == false &&
+        await SecurityManager().isDeviceDeveloperMode() == false;
+  }
+
+  Future checkDeviceSecurity(WidgetRef ref, BuildContext context) async {
+    if (Platform.isIOS == false && Platform.isAndroid == false) {
```

Comment:
Warning, the wallet should work on web platform too.
Could you add kIsWeb condition please

Reply:
We are only checking if the code is running in iOS or Android, to know if the device is jailbroken or in dev mode, we don't want to test if it is running in web in that case, right ?